api/src/controllers/provision.ts

```diff
@@ -89,15 +88,37 @@ export const provisionerCallbackHandler = async (
   }
 };
 
-const updateProvisionedProfile = async (profile: ProjectProfile): Promise<void> => {
+const updateProvisionedProfile = async (profile: ProjectProfile, clusterName: string): Promise<void> => {
+  const { RequestModel } = dm;
   try {
-    await updateProvisionStatus(profile, true);
+    // Must check if all bot_messages are complete before completing.
+    // Step 1. fetch active bot_messages for request
+
+    const requests = await RequestModel.findForProfile(Number(profile.id));
+    const request = requests.pop();
+    if (!request) {
+      return;
+    }
+
+    const botMessageSet = await fetchBotMessageRequests(Number(request.id))
+
+    if (botMessageSet.length === 1) {
```

Comment:
What's special about `1`? Would `> 0` or can we only ever have 1? Generally when I see numbers in code I think they're magic numbers.